src/Sylius/Bundle/CoreBundle/EventListener/Workflow/Order/SetImmutableNamesListener.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\EventListener\Workflow\Order;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Order\OrderItemNamesSetterInterface;
+use Symfony\Component\Workflow\Event\CompletedEvent;
+use Webmozart\Assert\Assert;
+
+final class SetImmutableNamesListener
+{
+    public function __construct(private OrderItemNamesSetterInterface $orderItemNamesSetter)
+    {
+    }
+
+    public function __invoke(CompletedEvent $event): void
+    {
+        /** @var OrderInterface $order */
+        $order = $event->getSubject();
+        Assert::isInstanceOf($order, OrderInterface::class);
+
+        $this->orderItemNamesSetter->__invoke($order);
```

Comment:
```suggestion
        $this->orderItemNamesSetter($order);
```

Reply:
Maybe im missing something but it doesnt work like that here :/, as i got error that that the orderItemNamesSetter method does not exisits